src/main/scala/cognite/spark/v1/MetricsSource.scala

```diff
@@ -2,22 +2,36 @@ package org.apache.spark.datasource
 
 import cats.Eval
 import com.codahale.metrics._
-import org.apache.spark._
-
 import java.util.concurrent.ConcurrentHashMap
+import org.apache.spark._
 
 class MetricsSource {
   // Add metricNamespace to differentiate with spark system metrics.
   // Keeps track of all the Metric instances that are being published
   val metricsMap = new ConcurrentHashMap[String, Eval[Counter]]
 
-  def getOrCreateCounter(metricNamespace: String, metricName: String): Counter = {
+  def getOrCreateCounter(metricNamespace: String, name: String): Counter = {
```

Comment:
```suggestion
  def getOrCreateDetailedCounter(metricNamespace: String, name: String): Counter = {
```
or something like that to indicate that meaning changes a lot while type signature remains the same `(String, String)->Counter`, this would enforce fixing references to it here or for external consumers

Reply:
Refactoring time